src/browser/findDOMNode.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2013-2014, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule findDOMNode
+ * @typechecks static-only
+ */
+
+"use strict";
+
+var ReactComponent = require('ReactComponent');
+var ReactInstanceMap = require('ReactInstanceMap');
+var ReactMount = require('ReactMount');
+
+var invariant = require('invariant');
+var isNode = require('isNode');
+
+/**
+ * Returns the DOM node rendered by this element.
+ *
+ * @param {ReactComponent|DOMElement} element
+ * @return {DOMElement} The root node of this element.
+ */
+function findDOMNode(componentOrElement) {
+  if (componentOrElement == null) {
+    return null;
+  }
+  if (isNode(componentOrElement)) {
+    return componentOrElement;
+  }
+  if (ReactInstanceMap.has(componentOrElement)) {
+    return ReactMount.getNodeFromInstance(componentOrElement);
+  }
+  invariant(
+    !(componentOrElement.render != null && typeof(componentOrElement.render) === 'function'),
```

Comment:
ok, done.  You guys seem to really like the inverted form of if statements.  Personally, I tend to think it's easier to understand "if the component has a render function, throw an error" rather than "assert that render is null *OR* that the render property is not a function".  The former is closer to the semantic reasoning, while the latter feels like ~~demorganized~~ **compiled** output.  The exclamation point out front just converts the "invariant" into an "if", and the inside of the "if" is the failure condition.  I know both are technically relying on short-circuit evaluation, but it just feels easier to reason about the former.  Anyway, that's my two cents.  Updated diff uses demorganized form.